### PR TITLE
Highlight distance snap grid rings that are close to the current time value

### PR DESCRIPTION
**Pull request by peppy**
This is intended to improve the usability of distance snap, especially for mappers coming across from stable (where distance snap is *only* available for the current point-in-time).

By highlighting the grid, it not only shows the correct point of placement if a mapper is looking to get the same behaviour as provided but stable, but should also help users understand exactly what the grid is showing in the first place. This is most noticeable when advancing time while the grid is visible:

https://user-images.githubusercontent.com/191335/167105984-65318479-5f25-4916-afd1-0a88de7e92d6.mp4

- [x] Depends on https://github.com/ppy/osu/pull/18103

### CODE REVIEW

**Comment by frenzibyte** [comment]
Looks pretty good, have pushed minor changes and one [fix](https://github.com/ppy/osu/pull/18117/commits/d78f1d158d3f5582fc99efd20a92c48e5e73331d) similar to https://github.com/ppy/osu/pull/18103/commits/246479bf34c725d886742017099f9d509e58b2f7.

**Review by frenzibyte — DISMISSED** [review]

File: osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs [hunk 1]
```diff
@@ -100,5 +102,40 @@ public override (Vector2 position, double time) GetSnappedPosition(Vector2 posit
...
+            }
+
+            protected override void Update()
+            {
+                base.Update();
+
+                if (editorClock == null)
+                    return;
+
+                float distanceForCurrentTime = snapProvider.DurationToDistance(referenceObject, editorClock.CurrentTime - referenceObject.GetEndTime());
```
Comment (peppy) [inline osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs:134]: Feel we may want to not pass the `referenceObject` in but instead just `StartTime`...

Comment (frenzibyte) [reply osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs:134]: Actually we'll still have to pass `referenceObject` either way, that's why I used `GetEndTime` instead. Either or works I think.

Comment (peppy) [reply osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs:134]: ah, let's leave it then

File: osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs [hunk 2]
```diff
@@ -100,5 +102,40 @@ public override (Vector2 position, double time) GetSnappedPosition(Vector2 posit
...
+                float timeBasedAlpha = 1 - Math.Clamp(Math.Abs(distanceForCurrentTime - Size.X / 2) / 30, 0, 1);
```
Comment (bdach) [inline osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs:135]: I don't think this is accounting for the distance spacing multiplier at all? Like this is what I'm seeing when I adjust the multiplier up and down while not changing time:

https://user-images.githubusercontent.com/20418176/167208952-df258fc2-1d93-4524-b284-f9fe37c7739f.mp4

I'm pretty sure the highlighted line shouldn't be shifting around like that, I'm not moving in time after all. Shouldn't this be something along the lines of

```suggestion
                float distanceForCurrentTime = (float)(snapProvider.DurationToDistance(referenceObject, editorClock.CurrentTime - referenceObject.GetEndTime()) * snapProvider.DistanceSpacingMultiplier.Value);
```

which yields the following?:

https://user-images.githubusercontent.com/20418176/167209011-309a04e3-6f0d-4ee2-a0e6-de21fdd3da96.mp4

Note that the highlighted ring does not change, it only expands and contracts (and that seems correct, I'm not moving in time after all).

**Review by bdach — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs (hunk 2 above)
Comment (frenzibyte) [reply osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs:135]: Good find, should've tested this well...

Comment (peppy) [reply osu.Game/Screens/Edit/Compose/Components/CircularDistanceSnapGrid.cs:135]: Looks like this broke in my refactor clean up efforts.

**Review by bdach — APPROVED** [review]

**Pull request merged**